apps/src/blockly/customBlocks/googleBlockly/spritelabBlocks.js

```diff
@@ -302,74 +302,84 @@ export const blocks = {
-            });
-            block.inputList[block.inputList.length - 1].appendField(
-              editButton,
-              'EDIT'
-            );
-            // getProcedureModel is defined on procedure blocks as part of
-            // @blockly/block-shareable-procedures
-            // For this block, we will get the procedure based on selected
-            // dropdown field option.
-            block.getProcedureModel = function () {
-              const fieldValue = block.getFieldValue(inputConfig.name);
-              const procedureMap = block.workspace.getProcedureMap();
-              let procedure = undefined;
-              for (const value of procedureMap.values()) {
-                if (value.name === fieldValue) {
-                  procedure = value;
-                  break;
-                }
-              }
-              return procedure;
-            };
+  addBehaviorPickerEditButton(
+    block,
+    inputConfig,
+    currentInputRow,
+    dropdownField
+  ) {
+    const behaviorsFound =
+      dropdownField.getOptions().length > 1 ||
+      dropdownField.getOptions()[0][1] !== NO_OPTIONS_MESSAGE;
+
+    // Behavior editing is only permitted using the modal function editor.
+    if (
+      behaviorsFound &&
+      Blockly.useModalFunctionEditor &&
+      // TODO: Support editing behaviors from within a modal editor workspace.
+      block.workspace.id === Blockly.getMainWorkspace().id
+    ) {
+      const editButton = new Blockly.FieldButton({
+        value: msg.edit(),
+        onClick: editButtonHandler,
+        colorOverrides: {button: 'blue', text: 'white'},
+      });
+      block.inputList[block.inputList.length - 1].appendField(
+        editButton,
+        'EDIT'
+      );
+      // getProcedureModel is defined on procedure blocks as part of
+      // @blockly/block-shareable-procedures
+      // For this block, we will get the procedure based on selected
+      // dropdown field option.
+      block.getProcedureModel = function () {
+        const fieldValue = block.getFieldValue(inputConfig.name);
+        const procedureMap = block.workspace.getProcedureMap();
+        let procedure = undefined;
+        for (const value of procedureMap.values()) {
+          if (value.name === fieldValue) {
+            procedure = value;
+            break;
           }
         }
-      },
-      generateCode(block, arg) {
-        const fieldValue = block.getFieldValue(arg.name);
-        const invalidBehavior = fieldValue === NO_OPTIONS_MESSAGE;
-        const behaviorId = Blockly.JavaScript.nameDB_?.getName(
-          fieldValue,
-          'PROCEDURE'
-        );
-        if (invalidBehavior) {
-          console.warn('No behaviors available');
-          return undefined;
-        } else {
-          return `new Behavior(${behaviorId}, [])`;
-        }
-      },
-    };
+        return procedure;
+      };
+    }
+  },
+  behaviorPickerGenerateCode(block, arg) {
```

Comment:
is this used?

Reply:
Nope! It was at one point, missed it in cleanup. Will remove.